sdks/python/apache_beam/io/vcfio_test.py

```diff
@@ -0,0 +1,451 @@
+  VCF_FILE_DIR_MISSING = not os.path.exists(get_full_dir())
+
+  def _create_temp_vcf_file(self, lines, tmpdir=None):
+    return self._create_temp_file(suffix='.vcf', tmpdir=tmpdir, lines=lines)
+
+  def _read_records(self, file_or_pattern):
+    source = VcfSource(file_or_pattern)
+    range_tracker = source.get_range_tracker(None, None)
+    read_data = [record for record in source.read(range_tracker)]
+    return read_data
+
+  def _create_temp_file_and_read_records(self, lines):
+    return self._read_records(self._create_temp_vcf_file(lines))
+
+  def _assert_variants_equal(self, actual, expected):
+    self.assertEqual(
+        sorted(expected, cmp=_variant_comparator),
+        sorted(actual, cmp=_variant_comparator))
+
+  def _get_sample_variant_1(self):
+    vcf_line = ('20	1234	rs123;rs2	C	A,T	50	PASS	AF=0.5,0.1;NS=1	'
+                'GT:GQ	0/0:48	1/0:20\n')
+    variant = Variant(
+        reference_name='20', start=1233, end=1234, reference_bases='C',
+        alternate_bases=['A', 'T'], names=['rs123', 'rs2'], quality=50,
+        filters=['PASS'],
+        info={'AF': VariantInfo(data=[0.5, 0.1], field_count='A'),
+              'NS': VariantInfo(data=1, field_count='1')})
+    variant.calls.append(
+        VariantCall(name='Sample1', genotype=[0, 0], info={'GQ': 48}))
+    variant.calls.append(
+        VariantCall(name='Sample2', genotype=[1, 0], info={'GQ': 20}))
+    return variant, vcf_line
+
+  def _get_sample_variant_2(self):
+    vcf_line = (
+        '19	123	rs1234	GTC	.	40	q10;s50	NS=2	GT:GQ	1|0:48	0/1:.\n')
+    variant = Variant(
+        reference_name='19', start=122, end=125, reference_bases='GTC',
+        alternate_bases=[], names=['rs1234'], quality=40,
+        filters=['q10', 's50'],
+        info={'NS': VariantInfo(data=2, field_count='1')})
+    variant.calls.append(
+        VariantCall(name='Sample1', genotype=[1, 0],
+                    phaseset=DEFAULT_PHASESET_VALUE,
+                    info={'GQ': 48}))
+    variant.calls.append(
+        VariantCall(name='Sample2', genotype=[0, 1], info={'GQ': None}))
+    return variant, vcf_line
+
+  def _get_sample_variant_3(self):
+    vcf_line = (
+        '19	12	.	C	<SYMBOLIC>	49	q10	AF=0.5	GT:GQ	0|1:45 .:.\n')
+    variant = Variant(
+        reference_name='19', start=11, end=12, reference_bases='C',
+        alternate_bases=['<SYMBOLIC>'], quality=49, filters=['q10'],
+        info={'AF': VariantInfo(data=[0.5], field_count='A')})
+    variant.calls.append(
+        VariantCall(name='Sample1', genotype=[0, 1],
+                    phaseset=DEFAULT_PHASESET_VALUE,
+                    info={'GQ': 45}))
+    variant.calls.append(
+        VariantCall(name='Sample2', genotype=[MISSING_GENOTYPE_VALUE],
+                    info={'GQ': None}))
+    return variant, vcf_line
+
+  @unittest.skipIf(VCF_FILE_DIR_MISSING, 'VCF test file directory is missing')
+  def test_read_single_file(self):
```

Comment:
<!--new_thread; commit:63511ddb8caf9e5c12888ef8a4d1071e0ea6c168; resolved:0-->
Can you add simple versions of the tests annotated with unittest.skipIf in this module ? I would like to have basic functionality tested instead of being skipped, whenever we run tests. Also rename these methods to test_read_single_file_large() and test_read_file_pattern_large() etc.

Reply:
Done.